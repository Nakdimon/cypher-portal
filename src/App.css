.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 85vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;  
}

.cypher-option {
  color: #282c34;
}

.cypher-select {
  height: 35px;
  margin: 15px 0px;
  border-radius: 5px; 
}

.cypher-input-text {
  height: 80px;
  width: 300px;
  border-radius: 5px;
  font-family: inherit;
  font: 400 13.3333px Arial;    
}

.App-footer {
  background-color: #282c34;
  min-height: 25vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;  
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
