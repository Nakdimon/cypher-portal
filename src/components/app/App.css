.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;  
}

.cypher-h1 {
  margin: 15px 0px;
}

.cypher-h3 {
  margin: 15px 0px;
}

.cypher-option {
  color: #282c34;
}

.cypher-select {
  height: 35px;
  margin: 15px 0px;
  border-radius: 5px; 
}

.cypher-input-text {
  height: 80px;
  width: 300px;
  border-radius: 5px;
  font-family: inherit;
  font: 400 13.3333px Arial;    
}
.button_cont {
  margin: 30px 0px 0px;
}

.cypher-button {
  color: #fff !important;
  font-size: 25px;
  text-transform: uppercase;
  text-decoration: none;
  background: #213764;
  padding: 10px;
  border-radius: 5px;
  display: inline-block;
  border: none;
  transition: all 0.4s ease 0s;
}

.disabled {
  color: currentColor;
  cursor: not-allowed;
  opacity: 0.5;
  text-decoration: none;
}

.cypher-button:hover {
  background: #434343;
  letter-spacing: 1px;
  -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  box-shadow: 5px 40px -10px rgba(0,0,0,0.57);
  transition: all 0.4s ease 0s;
  }

.App-footer {
  background-color: #282c34;
  min-height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;  
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
