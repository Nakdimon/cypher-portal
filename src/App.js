import React, { useState } from 'react';
import './App.css';
import cypherOptionList from './cypherOptionList'
import axios from 'axios';
import _debounce from 'lodash/debounce'

function App() {
  const [cypher, setCypher] = useState(null)

  const [cypheredText, setCypheredText] = useState('You\'ve yet to input something...')

  const[isTextAreaDisabled, setIsTextAreaDisabled] = useState(true)

  const[text, setText] = useState('default')
  
  const handleSelectChange = event => {
    setIsTextAreaDisabled(false)
    setCypher(event.target.value)
  }

  const handleTextAreaChange = _debounce((value) => {    
    setText(value)
    getCypheredText();
  },500)


  const getCypheredText = () => {
    axios.get('/'+cypher+'?text=')
    .then(function (response) {
      setCypheredText(response.data)
    }).catch(function(error) {
      setCypheredText('WHOOPS! A Mishap...')
    })
  }

  return (
    <div className="App">
      <header className="App-header">
        <h1>Cypher any text!</h1>
        <h3>Choose your cypher</h3>        
        <select className="cypher-select" onChange={handleSelectChange}>
          <option disabled selected>Select a cypher</option>
          {
            cypherOptionList.map((cypherOption) => 
            <option className="cypher-option" key={cypherOption.value} value={cypherOption.value}>{cypherOption.label}</option>)
          }
        </select>
        <textarea disabled={isTextAreaDisabled} className="cypher-input-text" onChange={handleTextAreaChange} placeholder="Input the text you want to cypher."></textarea>
        <h4>Cyphered Result</h4>
        <label className="cypher-result">{cypheredText}</label>
      </header>
      <footer className="App-footer"></footer>
    </div>
  );
}

export default App;
