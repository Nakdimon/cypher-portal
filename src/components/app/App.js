import React, { useState } from 'react';
import './App.css';
import cypherOptionList from '../assets/cypherOptionList.json'
import axios from 'axios';

function App() {
  const [cypher, setCypher] = useState(null)

  const [cypheredText, setCypheredText] = useState('You\'ve yet to input something...')

  const[isTextAreaDisabled, setIsTextAreaDisabled] = useState(true)
  
  const[isRequestButtonDisabled, setIsRequestButtonDisabled] = useState('disabled')

  const[text, setText] = useState('default')
  
  function handleSelectChange (event) {
    console.log(event.target.value)
    setIsTextAreaDisabled(false)
    setCypher(event.target.value)
  }

  function handleTextAreaChange (event) {
    if(event.target.value === '') {
      setCypheredText('Dont\'t you want to cypher something? :/')
      setIsRequestButtonDisabled('disabled')
    } else {
      setText(event.target.value)
      setIsRequestButtonDisabled('')
    }
  }

  function handleCypherAction (event) {
    getCypheredText()
  }

  const getCypheredText = () => {
    axios.get('/'+cypher+'?text='+text)
    .then(function (response) {
      setCypheredText(response.data)
    }).catch(function(error) {
      setCypheredText('WHOOPS! A Mishap...')
    })
  }

  return (
    <div className="App">
      <header className="App-header">
        <h1 className="cypher-h1">Cypher any text!</h1>
        <h3 className="cypher-h3">Choose your cypher</h3>        
        <select className="cypher-select" onChange={handleSelectChange}>
          <option disabled selected>Select a cypher</option>
          {
            cypherOptionList.map((cypherOption) => 
            <option className="cypher-option" key={cypherOption.value} value={cypherOption.value}>{cypherOption.label}</option>)
          }
        </select>
        <textarea disabled={isTextAreaDisabled} className="cypher-input-text" onChange={handleTextAreaChange} placeholder="Input the text you want to cypher."></textarea>
        <div className="button_cont" align="center"><a className={'cypher-button'+' '+isRequestButtonDisabled} onClick={handleCypherAction}>Cypher Text!</a></div>
        <h3>Cyphered Result:</h3>
        <label className="cypher-result">{cypheredText}</label>
      </header>
      <footer className="App-footer"></footer>
    </div>
  );
}

export default App;
