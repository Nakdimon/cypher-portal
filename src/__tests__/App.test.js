import React from 'react';
import App from '../components/app/App';
import {
  shallow, mount
} from '../enzyme'

import axios from 'axios'

jest.mock('axios')

describe("App component: ", () => {
  it("Should render.", () => {
    const component = shallow( < App /> )
    expect(component.exists()).toBe(true)
  })

  it("Should match the snapshot.", () => {
    const component = shallow( < App /> )
    expect(component).toMatchSnapshot();
  })
})

describe("Select child: ", () => {
  it("Should enable the textarea on select change.", ()=> {
    const component = mount(< App />)
    
    const event = {
      preventDefault() {},
      target: {
          value: 'event'
      }
    }
    component.find('select').simulate('change', event)
    expect(component.find('textarea').prop('disabled')).toBe(false)
  })
})

describe("TextArea child: ", () => {
  it("Should keep button disabled until changed.", () => {
    const component = mount(< App />)
    
    const event = {
      preventDefault() {},
      target: {
          value: 'event'
      }
    }

    expect(component.find('a.disabled').exists()).toBe(true)
    
    component.find('textarea').simulate('change', event)

    expect(component.find('a.disabled').exists()).toBe(false)
  })

  it("Should enable button when changed and disable it again if cleared.", () => {
    const component = mount(< App />)
    
    const event = {
      preventDefault() {},
      target: {
          value: 'event'
      }
    }

    const emptyEvent = {
      preventDefault() {},
      target: {
          value: ''
      }
    }

    component.find('textarea').simulate('change', event)
    
    expect(component.find('a.disabled').exists()).toBe(false)

    component.find('textarea').simulate('change', emptyEvent)

    expect(component.find('a.disabled').exists()).toBe(true)
    
  })
})

describe("Button child: ", () => {
  it('Should be disabled by default.', () => {
    const component = mount(< App />)

    expect(component.find('a.disabled').exists()).toBe(true)
  })

  it('Should change label if textarea input is not valid. ', () => {
    const component = mount(< App />)

    const invalidInputEvent = {
      preventDefault() {},
      target: {
          value: '1234'
      }
    }

    component.find('textarea').simulate('change', invalidInputEvent)

    component.find('a').simulate('click')

    expect(component.find('label.cypher-result').text()).toBe('That\'s not a valid string :/')

  })

  it('Should make request to server when input is valid ', () => {
    const getSpy = jest.spyOn(axios, 'get')
    const component = mount(< App />)
    
    
    const event = {
      preventDefault() {},
      target: {
          value: 'request'
      }
    }    

    component.find('textarea').simulate('change', event)
    
    component.find('a').simulate('click')

    expect(getSpy).toBeCalled()
  })
})