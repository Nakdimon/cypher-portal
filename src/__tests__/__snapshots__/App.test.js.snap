// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component:  Matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="cypher-h1"
    >
      Cypher any text!
    </h1>
    <h3
      className="cypher-h3"
    >
      Choose your cypher
    </h3>
    <select
      className="cypher-select"
      onChange={[Function]}
    >
      <option
        disabled={true}
        selected={true}
      >
        Select a cypher
      </option>
      <option
        className="cypher-option"
        value="caesar"
      >
        Casear
      </option>
      <option
        className="cypher-option"
        value="rot13"
      >
        ROT13
      </option>
    </select>
    <textarea
      className="cypher-input-text"
      disabled={true}
      onChange={[Function]}
      placeholder="Input the text you want to cypher."
    />
    <div
      align="center"
      className="button_cont"
    >
      <a
        className="cypher-button disabled"
        onClick={[Function]}
      >
        Cypher Text!
      </a>
    </div>
    <h3>
      Cyphered Result:
    </h3>
    <label
      className="cypher-result"
    >
      You've yet to input something...
    </label>
  </header>
  <footer
    className="App-footer"
  />
</div>
`;

exports[`App component:  Matches the snapshot. 1`] = `ShallowWrapper {}`;
